### PR TITLE
Assorted tweeks: Faster embedding plot, Stay in box select mode, More Images at start, Dataset path does not overflow 

### PR DESCRIPTION
**Pull request by PaulHax**

### CODE REVIEW

File: src/nrtk_explorer/app/core.py [hunk 1]
```diff
@@ -50,7 +52,7 @@ def __init__(self, server=None):
 
         known_args, _ = self.server.cli.parse_known_args()
         self.input_paths = known_args.dataset
-        self.state.current_dataset = str(Path(self.input_paths[0]).resolve())
```
Comment (jourdain) [inline src/nrtk_explorer/app/core.py:53]: don't you want to have absolute path that are properly formed?

Comment (jourdain) [reply src/nrtk_explorer/app/core.py:53]: This helped us fix some issue on Windows

File: src/nrtk_explorer/app/core.py [hunk 2]
```diff
@@ -96,25 +98,26 @@ def clear_hovered(**kwargs):
...
         self.on_dataset_change()
 
     def on_dataset_change(self, **kwargs):
-        # Reset cache
+        self.state.dataset_ids = []  # sampled images
         self.context.dataset = get_dataset(self.state.current_dataset, force_reload=True)
         self.state.num_images_max = len(self.context.dataset.imgs)
+        self.state.num_images = min(self.state.num_images_max, NUMBER_OF_IMAGES_DEFAULT)
+        # Force trigger resample_images() even if value is the same. We don't call resample_images directly to avoid calling twice if num_images changed.
+        self.state.dirty("num_images")
```
Comment (jourdain) [inline src/nrtk_explorer/app/core.py:114]: Technically you could call ` self.state.num_images = 0` just after `self.state.dataset_ids = []` and you will get the dirty effect automatically and it won't look strange.

Comment (jourdain) [reply src/nrtk_explorer/app/core.py:114]:
```python
self.state.num_images = 0 # Force refresh even if same number in new dataset
```

**Review by jourdain — APPROVED** [review]

File: src/nrtk_explorer/app/core.py (hunk 1 above)
Comment (PaulHax) [reply src/nrtk_explorer/app/core.py:53]: Gotcha, I put the `Path(ds).resolve()` step where the file access happens, `dataset.py`.  Alternatively, could front load the "validation" where dataset paths are ingested from command line.

File: src/nrtk_explorer/app/core.py (hunk 2 above)
Comment (PaulHax) [reply src/nrtk_explorer/app/core.py:114]: Putting `self.state.num_images = 0` before did trigger the change listener strangly.

Comment (jourdain) [reply src/nrtk_explorer/app/core.py:114]: why is that strange?

Comment (PaulHax) [reply src/nrtk_explorer/app/core.py:114]: Sorry, typo.  Did *not* trigger the change listener.

Comment (jourdain) [reply src/nrtk_explorer/app/core.py:114]: Let's chat... I want to make sure we are on the same page on what you did vs what I said. ;-)

**Pull request merged**